mempool/tree_mempool.go

```diff
@@ -39,45 +40,55 @@ type llrbMempool struct {
 	// Track whether we're rechecking txs.
 	// These are not protected by a mutex and are expected to be mutated in
 	// serial (ie. by abci responses which are called in serial).
-	recheckCursor *lElement    // next expected response
-	recheckEnd    llrb.NodeKey // re-checking stops here
+	recheckCursor *tElement    // next expected response
+	recheckEnd    tree.NodeKey // re-checking stops here
+}
+
+// NewTreeMempool returns a new mempool with the given configuration and connection to an application.
+func NewTreeMempool(
```

Comment:
this method doesn't need to be exported. in the end we only need `NewLLRBMempool`, `NewBTreeMempool` etc.